src/translations/indexMapper.js

```diff
@@ -0,0 +1,362 @@
+import { arrayFilter, arrayMap } from './../helpers/array';
+import { getListWithRemovedItems, getListWithInsertedItems } from './maps/utils/visuallyIndexed';
+import { rangeEach } from '../helpers/number';
+import IndexMap from './maps/indexMap';
+import SkipMap from './maps/skipMap';
+import MapCollection from './mapCollection';
+import localHooks from '../mixins/localHooks';
+import { mixin } from '../helpers/object';
+import { isDefined } from '../helpers/mixed';
+
+class IndexMapper {
+  constructor() {
+    /**
+     * Map storing the sequence of indexes.
+     *
+     * @type {IndexMap}
+     */
+    this.indexesSequence = new IndexMap();
```

Comment:
Inconsistency in naming. I'd propose to call the property `indexMap`.

Reply:
For me it's ok. We will discuss that.